### PR TITLE
Merge of pull request  #4706 - pg range handling improved to HEAD

### PR DESCRIPTION
**Pull request by yazshel**
I've merged @alekbarszczewski's pull request #4706 with master and resolved the conflicts The changes to the parse() method are greatly simplified with the recent master changes.

### CODE REVIEW

**Comment by alekbarszczewski** [comment]
@yazshel wow thanks man! I had busy days and couldn't look into it.

**Comment by mickhansen** [comment]
pg build is failing

File: lib/dialects/postgres/range.js [hunk 1]
```diff
@@ -37,7 +67,7 @@ function parse (value, parser) {
 
   result = result
     .map(function (value) {
-      return parser(value);
+      return parseRangeBound(parser);
```
Comment (janmeier) [inline lib/dialects/postgres/range.js]: I think you need to pass `value, parser` here

**Comment by yazshel** [comment]
Doh! Didn't test my code enough! I've committed dbfee88f58780a8c5087f3cb6640bf5d41c19f66 which also passes 'value'...

**Comment by yazshel** [comment]
Just to note, the conflict merge involved removing most of the date parsing changes from commit #2d02396 - these should no longer be necessary. AFAICT, the postgres dialect is now using getTypeParser() from the pg-native -> pg-types module, which handles parsing TIMESTAMP WITH TIMEZONE correctly.

**Comment by mickhansen** [comment]
PG build still failing

**Comment by yazshel** [comment]
I've added a commit updating the integration tests which should hopefully pass now... I haven't been able to test locally though because I'm on a Mac and the testing infrastructure is a PITA to set up.

One of the tests does require the `pg` module however, as it uses the type parser from this. I've made the test conditional on being able to load the module. Is this the correct way to do it, or should the test be removed given it's now using the `pg` / `pg-types` modules to do the stuff that we're testing?

**Comment by janmeier** [comment]
It should not be necessary to check for the pg module, since we already do ` if (dialect.match(/^postgres/)) {` in the beginning of the file

**Comment by janmeier** [comment]
If you just remove the try around pg (move it inside the if instead), rebase and add a changelog entry this should be good to go

**Comment by yazshel** [comment]
@janmeier Rebased to `master` and changelog entry added.

**Comment by janmeier** [comment]
Thanks a lot @yazshel  and @alekbarszczewski  :)  !

**Comment by alekbarszczewski** [comment]
@janmeier / @yazshel Thank you!

**Comment by yazshel** [comment]
The thanks should go to you @alekbarszczewski - I'd been playing with [sequelize](https://github.com/sequelize/sequelize) for a little while and was disappointed with the missing support for infinite bounds in pg ranges... until I discovered your pull request! Keep up the great work!
Thanks @janmeier for your advice and the pull.

**Pull request merged**